### PR TITLE
Add tag filter to Search Screen

### PR DESCRIPTION
**Pull request by hoapq-1005**
## Related Tickets

- [#26689](https://edu-redmine.sun-asterisk.vn/issues/26689)

## What's this PR do ?

- [x] Add list current tags to improve UX for Search Screen

## Library
*(List gem, library third party add new)*

- MaterialComponents/Chips, 110.3.0

## Impacted Areas in Application
*(List features, api, models or services that this PR will affect)*

- Scene/Search

## Checklist

- [x] It was tested in local success ?
- [x] Fill link PR into ticket and the opposite
- [x] Note reason, scope of influence, solution into ticket
- [x] Validate UI/Model/API

## Screenshot
image

### CODE REVIEW

File: Delicious/Scenes/Search/SearchViewController.swift [hunk 1]
```diff
@@ -172,26 +194,32 @@ final class SearchViewController: UIViewController, BindableType {
...
             .disposed(by: rx.disposeBag)
         output.searched
-            .do(onNext: showResults)
+            .do(onNext: { [weak self] in
+                if $0.query.isEmpty && $0.tags.isEmpty {
+                    self?.showTags()
+                } else {
+                    self?.showResults()
+                }
+            })
```
Comment (dongtg-1485) [inline Delicious/Scenes/Search/SearchViewController.swift]: cho nó thành hẳn binding property đi em, ko nên quá lạm dụng side effect

File: Delicious/Scenes/Search/SearchViewController.swift [hunk 2]
```diff
@@ -92,24 +96,39 @@ final class SearchViewController: UIViewController, BindableType {
...
+
         let suggestDataSource = RxTableViewSectionedReloadDataSource<AutoCompletionSection>(configureCell: { (_, tableView, indexPath, item) -> UITableViewCell in
             return tableView.dequeueReusableCell(
                 for: indexPath,
                 cellType: SuggestTableViewCell.self)
-            .then {
-                $0.setUp(item)
+                .then {
+                    $0.setUp(item)
             }
```
Comment (sonnd-1148) [inline Delicious/Scenes/Search/SearchViewController.swift:151]: tab

File: Delicious/Scenes/Search/SearchViewModel.swift [hunk 3]
```diff
@@ -63,8 +74,8 @@ extension SearchViewModel: ViewModelType {
 
         let (page, error, isLoading, isReloading, isLoadingMore) = getSearch.destructured
 
-        let results = page.map { data in
-            self.useCase.getSearchResultSection(data: data.items)
+        let results = page.withLatestFrom(searchModel) { (page, model) in
```
Comment (sonnd-1148) [inline Delicious/Scenes/Search/SearchViewModel.swift]: mấy cái operator nên xuống dòng cho nó dễ nhìn nhé

File: Delicious/Scenes/Search/SearchViewModel.swift [hunk 4]
```diff
@@ -49,9 +50,19 @@ extension SearchViewModel: ViewModelType {
             input.searchTrigger.map { SearchModel(query: $0, tags: []) },
             input.tagTrigger.map { SearchModel(query: "", tags: [$0]) }
         )
+        
+        let removeTag = input.removeTagTrigger.withLatestFrom(searchModel) { tag, model  -> SearchModel in
```
Comment (sonnd-1148) [inline Delicious/Scenes/Search/SearchViewModel.swift]: đây cũng thế. break line

**Review by sonnd-1148 — APPROVED** [review]

File: Delicious/Scenes/Search/SearchViewController.swift [hunk 5]
```diff
@@ -133,16 +166,18 @@ final class SearchViewController: UIViewController, BindableType {
...
+            removeTagTrigger: removeTagTrigger.asDriverOnErrorJustComplete()
         )
         let output = viewModel.transform(input)
-        
+
         suggestTableView.rx
             .modelSelected(String.self)
             .do(onNext: {
                 self.searchBar.text = $0
                 self.searchTrigger.onNext($0)
```
Comment (datdt-1935) [inline Delicious/Scenes/Search/SearchViewController.swift]: retain cycle này em

**Review by datdt-1935 — APPROVED** [review]

**Pull request merged**